### PR TITLE
Added SessionToggle, SessionModal, and SessionDropdown comp…

### PR DESCRIPTION
**Pull request by vincentbavitz**
Far from finished, but on the way to completing these three components for direct drop-in replacements for old elements.

### CODE REVIEW

File: ts/components/session/SessionToggle.tsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+      >
+        <div className="knob" />
+      </div>
+    );
+  }
+
+  private clickHandler() {
+    this.setState({
+      active: !this.state.active,
+    });
```
Comment (Bilb) [inline ts/components/session/SessionToggle.tsx]: this element does not give back the onclick event for now. We will need it as soon as we use it somewhere.

File: js/background.js [hunk 2]
```diff
@@ -802,6 +802,18 @@
...
+    //     el: $('#settings-container'),
+    //   });
+    //   window.settingsView.render();
+    // });
+
+    window.generateID = () =>
+      Math.random()
+        .toString(36)
+        .substring(3);
+
```
Comment (Bilb) [inline js/background.js]: We should either keep this one or the one in ComponentTools.tsx

File: ts/components/session/SessionToggle.tsx (hunk 1 above)
Comment (vincentbavitz) [reply ts/components/session/SessionToggle.tsx]: Amended

File: js/background.js (hunk 2 above)
Comment (vincentbavitz) [reply js/background.js]: Resolved

**Pull request merged**